rust/index/src/spann/types.rs

```diff
@@ -167,6 +168,88 @@ impl Configurable<RandomSamplePolicyConfig> for RandomSamplePolicy {
+    pub versions_map_flush_latency: opentelemetry::metrics::Histogram<u64>,
+    pub hnsw_flush_latency: opentelemetry::metrics::Histogram<u64>,
+    pub num_pl_blocks_flushed: opentelemetry::metrics::Counter<u64>,
+    pub num_versions_map_blocks_flushed: opentelemetry::metrics::Counter<u64>,
+}
+
+impl Default for SpannMetrics {
+    fn default() -> Self {
+        let meter = global::meter("chroma");
+        let num_pl_modified = meter.u64_counter("num_pl_modified").build();
+        let num_heads_created = meter.u64_counter("num_heads_created").build();
+        let num_heads_deleted = meter.u64_counter("num_heads_deleted").build();
+        let num_reassigns = meter.u64_counter("num_reassigns").build();
+        let num_splits = meter.u64_counter("num_splits").build();
+        let num_reassigns_split_point = meter.u64_counter("num_reassigns_split_point").build();
+        let num_reassigns_nbrs = meter.u64_counter("num_reassigns_nbrs").build();
+        let num_reassigns_merged_point = meter.u64_counter("num_reassigns_merged_point").build();
+        let num_centers_fetched_rng = meter.u64_counter("num_centers_fetched_rng").build();
+        let pl_commit_latency = meter
+            .u64_histogram("pl_commit_latency")
+            .with_unit("ms")
+            .build();
+        let versions_map_commit_latency = meter
+            .u64_histogram("versions_map_commit_latency")
+            .with_unit("ms")
+            .build();
+        let hnsw_commit_latency = meter
+            .u64_histogram("hnsw_commit_latency")
+            .with_unit("ms")
+            .build();
+        let pl_flush_latency = meter
+            .u64_histogram("pl_flush_latency")
+            .with_unit("ms")
+            .build();
+        let versions_map_flush_latency = meter
+            .u64_histogram("versions_map_flush_latency")
+            .with_unit("ms")
+            .build();
+        let hnsw_flush_latency = meter
+            .u64_histogram("hnsw_flush_latency")
+            .with_unit("ms")
+            .build();
+        let num_pl_blocks_flushed = meter.u64_counter("num_pl_blocks_flushed").build();
+        let num_versions_map_blocks_flushed =
+            meter.u64_counter("num_versions_map_blocks_flushed").build();
+        Self {
+            num_pl_modified,
+            num_heads_created,
+            num_heads_deleted,
+            num_reassigns,
+            num_splits,
+            num_reassigns_split_point,
+            num_reassigns_nbrs,
+            num_reassigns_merged_point,
+            num_centers_fetched_rng,
+            pl_commit_latency,
+            versions_map_commit_latency,
+            hnsw_commit_latency,
+            pl_flush_latency,
+            versions_map_flush_latency,
+            hnsw_flush_latency,
+            num_pl_blocks_flushed,
+            num_versions_map_blocks_flushed,
+        }
+    }
+}
+
 #[derive(Clone)]
 // Note: Fields of this struct are public for testing.
```

Comment:
In general, in another PR to reduce scope of this one, it would be good to split this file up somewhat.